packages/iris-grid/src/IrisGridPartitionSelector.tsx

```diff
@@ -1,316 +1,394 @@
+  data: {
+    [key: string]: unknown;
+  };
+}
+
+interface IrisGridPartitionSelectorProps {
   dh: DhType;
-  getFormattedString: (value: T, type: string, name: string) => string;
-  table: Table;
-  columns: Column[];
-  partitions: (string | null)[];
-  onFetchAll: () => void;
-  onDone: (event?: React.MouseEvent<HTMLButtonElement>) => void;
-  onChange: (partitions: (string | null)[]) => void;
+  getFormattedString: (value: unknown, type: string, name: string) => string;
+  tablePromise: Promise<Table>;
+  columns: readonly Column[];
+  partitionConfig: PartitionConfig;
+  onMerge: () => void;
+  onChange: (partitionConfig: PartitionConfig) => void;
+  onKeyTable: () => void;
 }
 interface IrisGridPartitionSelectorState {
-  partitions: (string | null)[];
-  partitionTables: Table[] | null;
+  partitionConfig: PartitionConfig;
+  partitionColumnValues: readonly Items[];
+  selectorValue: readonly string[];
 }
-class IrisGridPartitionSelector<T> extends Component<
-  IrisGridPartitionSelectorProps<T>,
+class IrisGridPartitionSelector extends Component<
+  IrisGridPartitionSelectorProps,
   IrisGridPartitionSelectorState
 > {
   static defaultProps = {
     onChange: (): void => undefined,
-    onFetchAll: (): void => undefined,
-    onDone: (): void => undefined,
+    onMerge: (): void => undefined,
+    onKeyTable: (): void => undefined,
     partitions: [],
   };
 
-  constructor(props: IrisGridPartitionSelectorProps<T>) {
+  constructor(props: IrisGridPartitionSelectorProps) {
     super(props);
 
-    this.handleCloseClick = this.handleCloseClick.bind(this);
-    this.handleIgnoreClick = this.handleIgnoreClick.bind(this);
-    this.handlePartitionChange = this.handlePartitionChange.bind(this);
+    this.handleKeyTableClick = this.handleKeyTableClick.bind(this);
+    this.handleMergeClick = this.handleMergeClick.bind(this);
     this.handlePartitionSelect = this.handlePartitionSelect.bind(this);
-    this.handlePartitionListResized =
-      this.handlePartitionListResized.bind(this);
-    this.handleSearchOpened = this.handleSearchOpened.bind(this);
-    this.handleSearchClosed = this.handleSearchClosed.bind(this);
 
-    const { dh, columns, partitions } = props;
+    const { dh, columns, partitionConfig } = props;
     this.tableUtils = new TableUtils(dh);
-    this.searchMenu = columns.map(() => null);
-    this.selectorSearch = columns.map(() => null);
+    this.table = null;
+    this.partitionTables = null;
 
     this.state = {
-      partitions,
-      partitionTables: null,
+      selectorValue: columns.map(() => ''),
+      partitionConfig,
+      partitionColumnValues: columns.map(() => ({ order: [], data: {} })),
     };
   }
 
   async componentDidMount(): Promise<void> {
-    const { columns, table } = this.props;
-    const { partitions } = this.state;
+    const { columns, partitionConfig, tablePromise } = this.props;
 
-    const partitionTables = await Promise.all(
+    const table = await tablePromise;
```

Comment:
As I commented in IrisGrid, pass in the `model` instead of a `tablePromise`, and load the table from the model here.